hooks/useSavedGuideIds.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import { SAVED_GUIDES_LOCAL_STORAGE } from "utils/constants";
+
+function getGuideIds() {
+  const rawSavedGuideIds = localStorage.getItem(SAVED_GUIDES_LOCAL_STORAGE);
+
+  const savedGuideIds: string[] = (() => {
+    try {
+      return JSON.parse(rawSavedGuideIds || "[]") || [];
+    } catch (err) {
+      console.error(
+        "Problem parsing shared guides, returning empty array...",
+        err
+      );
+      return [];
+    }
+  })();
+
+  return savedGuideIds;
+}
+
+export function useSavedGuideIds() {
```

Comment:
There's some benefit in defining a `ReturnType` for the hook itself to infer directly what it provides.

Reply:
done